### PR TITLE
CLI: ipinfo tool prefix addr

### PR DESCRIPTION
**Pull request by AhmadMujahid2k**

### CODE REVIEW

File: ipinfo/cmd_tool_prefix.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		"--help": predict.Nothing,
+	},
+}
+
+func printHelpToolPrefix() {
+	fmt.Printf(
+		`Usage: %s tool prefix <cmd> [<opts>] [<args>]
+
+Commands:
+	addr	returns prefix's IP address.
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_prefix.go:27]: Note to not use real tab chars

Comment (UmanShahzad) [reply ipinfo/cmd_tool_prefix.go:27]: Also the help message can be a bit more specific, what IP address exactly?

File: ipinfo/cmd_tool_prefix_addr.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		"--help": predict.Nothing,
+	},
+}
+
+func printHelpToolPrefixAddr() {
+	fmt.Printf(
+		`Usage: %s tool prefix addr <cidr>
+
+Description:
+	returns prefix's IP address.
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_prefix_addr.go:24]: Please dont use real tab chars in help msgs and also same comment here about a better description for this command.

File: lib/cmd_tool_prefix_addr.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+	op := func(input string, inputType INPUT_TYPE) error {
+		switch inputType {
+		case INPUT_TYPE_CIDR:
+			prefix, err := netip.ParsePrefix(input)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("Address: %s\n", prefix.Addr())
+		default:
```
Comment (UmanShahzad) [inline lib/cmd_tool_prefix_addr.go:36]: We can simply kill the default case, it'll skip over bad values

File: lib/cmd_tool_prefix_addr.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	}
+
+	op := func(input string, inputType INPUT_TYPE) error {
+		switch inputType {
+		case INPUT_TYPE_CIDR:
+			prefix, err := netip.ParsePrefix(input)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("Address: %s\n", prefix.Addr())
```
Comment (UmanShahzad) [inline lib/cmd_tool_prefix_addr.go:35]: Should be like the default case - print the input and the IP on the other side

File: ipinfo/cmd_tool_prefix.go (hunk 1 above)
Comment (AhmadMujahid2k) [reply ipinfo/cmd_tool_prefix.go]: so, it just return the IP address of p (here p is Prefix Struct type) for example if the input is = 8.8.8.8/12 output will be 8.8.8.8

File: ipinfo/cmd_tool_prefix.go [hunk 5]
```diff
@@ -24,7 +24,7 @@ func printHelpToolPrefix() {
 		`Usage: %s tool prefix <cmd> [<opts>] [<args>]
 
 Commands:
-	addr	returns prefix's IP address.
+	addr  returns IP address of CIDR using prefix.
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_prefix.go:27]: Still using a tab here

File: ipinfo/cmd_tool_prefix_addr.go [hunk 6]
```diff
@@ -21,7 +21,7 @@ func printHelpToolPrefixAddr() {
 		`Usage: %s tool prefix addr <cidr>
 
 Description:
-	returns prefix's IP address.
+ returns IP address of CIDR using prefix.
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_prefix_addr.go:24]: Using wrong number of spaces

Also let us make the description just "returns the base IP address of a prefix."

File: ipinfo/cmd_tool_prefix.go (hunk 5 above)
Comment (AhmadMujahid2k) [reply ipinfo/cmd_tool_prefix.go:27]: it just 2 space bar.

File: ipinfo/cmd_tool_prefix_addr.go (hunk 6 above)
Comment (AhmadMujahid2k) [reply ipinfo/cmd_tool_prefix_addr.go:24]: ok sure

File: ipinfo/cmd_tool_prefix_addr.go [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+		"--help": predict.Nothing,
+	},
+}
+
+func printHelpToolPrefixAddr() {
+	fmt.Printf(
+		`Usage: %s tool prefix addr <cidr>
+
+Description:
+  returns the base IP address of a prefix.
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_prefix_addr.go:24]:
```suggestion
  Returns the base IP address of a prefix.
```

**Review by UmanShahzad — APPROVED** [review]

**Pull request merged**